crates/edr_evm/src/block/builder.rs

```diff
@@ -94,7 +95,8 @@ where
         >,
         state: Box<dyn SyncState<Self::StateError>>,
         cfg: CfgEnv<ChainSpecT::Hardfork>,
-        options: BlockOptions,
+        withdrawals: Option<Vec<Withdrawal>>,
```

Comment:
IIUC, this will only ever be used for L1. Does it make sense to add it to the `BlockBuilder` trait that will need to be used for all chain types? I'm wondering if this should be specific to the `ChainSpecT` somehow...

Reply:
I tried to adapt this but as expressed in this [comment](https://github.com/NomicFoundation/edr/pull/977#discussion_r2183195009), it requires larger refactors in the provider to be able to determine which inputs should be sent to which block builder.